tests/cycle_initial_call_query.rs

```diff
@@ -0,0 +1,35 @@
+//! It's possible to call a Salsa query from within a cycle initial fn.
```

Comment:
Another test idea: A recovery or initial function that calls into the cycle itself.

Reply:
Added some tests for this. I'm not sure to what extent we want to promise any specific behavior here; in the docs I expect to just say "don't do this." But the behavior that falls out is actually somewhat reasonable, and it makes sense to have tests for that case.